mkosi/distributions/serpentos.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import shutil
+from collections.abc import Sequence
+
+from mkosi.config import Architecture
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer.moss import invoke_moss
+from mkosi.log import die
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "Serpent OS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "btrfs"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.moss
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "rolling"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.serpentos
+
+    @classmethod
+    def tools_tree_packages(cls) -> list[str]:
+        return [
+            "dash", # for sh
+            "systemd",
+            "zstd",
+        ]
+
+    @classmethod
+    def setup(cls, context: Context) -> None:
+        if not shutil.which("moss"):
+            die(f"moss binary not found but needed to install {cls.pretty_name()}")
+
+        # needed to normalize modification time of /boot later:
+        (context.root / "boot").mkdir(exist_ok=True)
+
+    @classmethod
+    def install(cls, context: "Context") -> None:
+        pass
```

Comment:
So in `install()` we have to install the basic filesystem packages. Please check the other distributions for examples. Does Serpent OS have a package that provides basic directories such as /usr and /etc? We should also install the package that provides /etc/passwd and friends here.

Reply:
@hunger Right but when does it run systemd-tmpfiles and such to populate the rest of the system? Does it intend to run these all at firstboot or are these run by the package manager?

Reply:
@hunger Still need an answer to my question here.

Reply:
Right just leave it empty for now we can figure this out later.